src/librustc_resolve/diagnostics.rs

```diff
@@ -769,6 +769,11 @@ impl<'a> Resolver<'a> {
         span: Span,
     ) -> bool {
         if let Some(suggestion) = suggestion {
+            // We shouldn't suggest underscore.
+            if suggestion.candidate.as_str() == "_" {
```

Comment:
You could compare against [`kw::Underscore`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/symbol/kw/constant.Underscore.html) instead, i.e. `if suggestion.candidate == kw::Underscore  { ... }`.